server/src/storage/mod.rs

```diff
@@ -167,6 +183,103 @@ impl StorageEngine {
         }
     }
 
+    /// Return the record batch schema used for uncompressed segments.
+    pub fn get_uncompressed_segment_schema() -> Schema {
+        Schema::new(vec![
+            Field::new("timestamps", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("values", ArrowValue::DATA_TYPE, false),
+        ])
+    }
+
+    /// Return the record batch schema used for compressed segments.
+    pub fn get_compressed_segment_schema() -> Schema {
+        Schema::new(vec![
+            Field::new("model_type_id", DataType::UInt8, false),
+            Field::new("timestamps", DataType::Binary, false),
+            Field::new("start_time", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("end_time", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("values", DataType::Binary, false),
+            Field::new("min_value", ArrowValue::DATA_TYPE, false),
+            Field::new("max_value", ArrowValue::DATA_TYPE, false),
+            Field::new("error", DataType::Float32, false),
+        ])
+    }
+
+    // TODO: Test using more efficient encoding. Plain encoding makes it easier to read the files externally.
+    /// Write `batch` to an Apache Parquet file at the location given by `path`. Return Ok if the
+    /// file was written successfully, otherwise return `ParquetError`.
+    pub fn write_batch_to_apache_parquet_file(
+        batch: RecordBatch,
+        path: &Path,
+    ) -> Result<(), ParquetError> {
+        let error = ParquetError::General(
+            format!("Apache Parquet file at path '{}' could not be created.", path.display())
+        );
+
+        // Check if the extension of the given path is correct.
+        if path.extension().and_then(OsStr::to_str) == Some("parquet") {
+            if let Ok(file) = File::create(path) {
```

Comment:
I was actually looking into this briefly since I did not like the two `else` blocks right after each other and large indentation but i was not able to find a fix. `map_err()` looks good and i have replaced the `if else` with it.